.github/test-shards.py

```diff
@@ -0,0 +1,101 @@
+# Shard detection script for GitHub Actions
```

Comment:
why python and not Javascript?

Reply:
Otherwise, we need to run `setup-node` and, in some cases, await for `yarn install`, while `python` is ready to run

Reply:
Actually, I'm not sure, node should be available by default too

Reply:
I think we actually should use our `setup-node` action to make sure the cache is populated, otherwise lots of small actions will run `yarn install` on each cache miss. Running `yarn install` in the `setup` job isn't 100% guarantee, but in 99% cases it will work well I think. So yeah, I have a nice code rewriting task for Copilot chat.